Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -38,69 +42,102 @@ internal static void InstallProjectWideActions()
             GetOrCreate();
         }
 
-        internal static InputActionAsset GetOrCreate()
+        internal static InputActionAsset LoadFromAsset()
+        {
+            string text;
+            try
+            {
+                text = File.ReadAllText(assetPath);
+            }
+            catch
+            {
+                return null;
+            }
+
+            var asset = ScriptableObject.CreateInstance<InputActionAsset>();
+            asset.name = kAssetName;
+            try
+            {
+                asset.LoadFromJson(text);
+            }
+            catch (Exception exception)
+            {
+                Debug.LogError($"InputSystem could not parse input actions in JSON format from '{assetPath}' ({exception})");
+                Object.DestroyImmediate(asset);
+                return null;
+            }
+            return asset;
+        }
+
+        internal static InputActionAsset LoadFromInputManagerSettings()
         {
-            var objects = AssetDatabase.LoadAllAssetsAtPath(s_AssetPath);
+            var objects = AssetDatabase.LoadAllAssetsAtPath(kInputManagerAssetPath);
             if (objects != null)
             {
                 var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
                 if (inputActionsAsset != null)
                     return inputActionsAsset;
             }
+            return null;
+        }
 
+        internal static InputActionAsset GetOrCreate()
+        {
+            InputActionAsset asset;
+
+            asset = LoadFromAsset();
+            if (asset != null) return asset;
+
+            // v1.8.0-pre1 stored the Actions in ProjectSettings/InputManager.asset.
+            // Check if we have a project that was saved on that version and migrate it
+            asset = LoadFromInputManagerSettings();
+            if (asset != null)
+            {
+                // Migrate the pre1 asset
+                SaveAsset(asset);
+
+                // Load from the newly migrated asset.
+                asset = LoadFromAsset();
+                if (asset != null) return asset;
+            }
+
+            // Create a new one if we couldn't find any existing one to load.
             return CreateNewActionAsset();
         }
 
-        private static InputActionAsset CreateNewActionAsset()
+        static void SaveAsset(InputActionAsset asset)
         {
-            var json = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, s_DefaultAssetPath));
-
-            var asset = ScriptableObject.CreateInstance<InputActionAsset>();
-            asset.LoadFromJson(json);
-            asset.name = kAssetName;
+            try
+            {
+                if (!Directory.Exists(kAssetDirectory))
+                    Directory.CreateDirectory(kAssetDirectory);
 
-            AssetDatabase.AddObjectToAsset(asset, s_AssetPath);
+                var assetJson = asset.ToJson();
```

Comment:
I would recommend guarding this save with a comparison with existing content of that file to avoid touching the file if there is no content diff of the file. This would eliminate accidentally touching the file if there is no content diff (modifying timestamp of the file header)

Reply:
(Since a touch without content change would trigger unnecessary code generation)

Reply:
thanks, something to look at for sure 👍 
I just noticed this branch should be going to `develop`, this is cut off from the type safe API branch, let me just fix that.

Reply:
Changed the base branch of the PR now. Sorry for the confusion. But the code you pointed that probably needs to be looked up anyway 👍

Reply:
ThaNKS, I checked the underlying utility function and its already checking what I said it seems